packages/gestalt/src/Masonry/defaultTwoColumnModuleLayout.js

```diff
@@ -634,10 +634,10 @@ const defaultTwoColumnModuleLayout = <T: { +[string]: mixed }>({
         heights: $ReadOnlyArray<number>,
         positions: $ReadOnlyArray<{ item: T, position: Position }>,
       } = batches.reduce(
-        (acc, itemsToPosition, i) =>
+        (acc, itemsInBatch, i) =>
           getPositionsWithMultiColumnItem({
             multiColumnItem: multiColumnItems[i],
-            itemsToPosition,
+            itemsToPosition: itemsInBatch.filter((item) => !positionCache.has(item)),
```

Comment:
Although this works I think the main problem is that we are not filtering the positioned items when we generate the batches, I think it makes more sense just to change `items` to `itemsWithoutPositions` on the batches logic.

```
      const batchSize = multiColumnItems.length;
      const batches = Array.from({ length: batchSize }, (): $ReadOnlyArray<T> => []).map(
        (batch, i) => {
          const startIndex = i === 0 ? 0 : itemsWithoutPositions.indexOf(multiColumnItems[i]);
          const endIndex =
            i + 1 === multiColumnItems.length
              ? itemsWithoutPositions.length
              : itemsWithoutPositions.indexOf(multiColumnItems[i + 1]);
          return itemsWithoutPositions.slice(startIndex, endIndex);
        },
      );
```